tests/integration/test_apigateway.py

```diff
@@ -193,6 +193,13 @@ def test_create_rest_api_with_custom_id(self, apigateway_client, create_rest_api
         assert test_id == api_id
         assert apigw_name == name
 
+        spec_file = load_file(TEST_IMPORT_MOCK_INTEGRATION)
+        apigateway_client.put_rest_api(restApiId=test_id, body=spec_file, mode="overwrite")
+
+        url = path_based_url(api_id=test_id, stage_name="latest", path="/echo/foobar")
```

Comment:
Great idea to add a mock integration and add the invocation here! 👍 

Should we maybe test it for host-based URLs as well? (as that is what the reported issue is referring to)

We could for example parameterize the test method, as we also do in a few other places in this file:
```
@pytest.mark.parametrize("url_function", [path_based_url, host_based_url])
...
```